controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -229,7 +235,7 @@ func mergeStorageProperties(
 	configKey := types.NamespacedName{Namespace: namespace, Name: medusaSpec.MedusaConfigurationRef.Name}
 	if err := remoteClient.Get(ctx, configKey, storageProperties); err != nil {
 		logger.Error(err, fmt.Sprintf("failed to get MedusaConfiguration %s", configKey))
-		return result.Error(err)
+		return result.RequeueSoon(r.DefaultDelay)
```

Comment:
question: Why the change? The only difference I can see is the change of delay when reconcile happens (error will do a reconcile also - but with a backoff timer to prevent flooding the server which this doesn't).

Reply:
issue: This error handling is still incorrect, in case of error, err should be returned for proper requeue process (especially in this case when it can be Kubernetes API server overload or similar and we don't want to make it worse)